unity-renderer/Assets/DCLServices/CustomNftCollection/WebInterfaceCustomNftCollectionBridge.cs

```diff
@@ -0,0 +1,97 @@
+using Cysharp.Threading.Tasks;
+using DCL;
+using DCL.Helpers;
+using DCL.Interface;
+using JetBrains.Annotations;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using UnityEngine;
+
+namespace DCLServices.CustomNftCollection
+{
+    public class WebInterfaceCustomNftCatalogBridge : MonoBehaviour, ICustomNftCollectionService
+    {
+        private UniTaskCompletionSource<IReadOnlyList<string>> getCollectionsTask;
+        private UniTaskCompletionSource<IReadOnlyList<string>> getItemsTask;
+
+        public static WebInterfaceCustomNftCatalogBridge GetOrCreate()
+        {
+            var bridgeObj = SceneReferences.i?.bridgeGameObject;
+
+            return SceneReferences.i?.bridgeGameObject == null
+                ? new GameObject("Bridge").AddComponent<WebInterfaceCustomNftCatalogBridge>()
+                : bridgeObj.GetOrCreateComponent<WebInterfaceCustomNftCatalogBridge>();
+        }
+
+        public void Dispose()
+        {
+        }
+
+        public void Initialize()
+        {
+        }
+
+        public async UniTask<IReadOnlyList<string>> GetConfiguredCustomNftCollectionAsync(CancellationToken cancellationToken)
+        {
+            try
+            {
+                if (getCollectionsTask != null)
+                    return await getCollectionsTask.Task.AttachExternalCancellation(cancellationToken);
+
+                getCollectionsTask ??= new UniTaskCompletionSource<IReadOnlyList<string>>();
+
+                WebInterface.GetWithCollectionsUrlParam();
+
+                return await getCollectionsTask.Task
+                                                .Timeout(TimeSpan.FromSeconds(30))
+                                                .AttachExternalCancellation(cancellationToken);
+            }
+            catch (Exception)
```

Comment:
If i ignore it then the task (`getCollectionsTask`) will stay != null, so the next time i use it, its going to return a cancelled task

Reply:
Its true, but I think its not OK to ignore `OperationCanceledException` at this level. Instead its better to handle it from the client, in this case the UI. Moreover it will require me to return an invalid list, which makes no sense in the current flow

Reply:
But suggestion is that you don't nullify `getCollectionsTask = null;` as cancellation has nothing to do with the exception in the task itself

Reply:
I had to rework the approach. Using a UniTaskCompletionSource<IReadOnlyList<string>> was emitting an invalid string array on WebGL (was working on on desktop&editor). Seems like a IL2CPP issue, not sure. This workaround is the only one i found to make it work on all platforms. I dont like it, but works.
This call caused the issue:
```
IReadOnlyList<string> customCollections = await customNftCollectionService.GetConfiguredCustomNftCollectionAsync(cancellationToken);
```

Here were the logs which made no sense to me:
When logging the array. Not even logging the type of it, just nothing:
```
BackpackEmotesSectionController.FetchCustomEmoteCollections: 
```

When trying to iterate:
```
Not implemented: Class::FromIl2CppType

System.MethodAccessException: Attempt to access method 'System.Collections.Generic.IEnumerable<System.String>.GetEnumerator' on type '' failed. 

MethodAccessException: Attempt to access method 'System.Collections.Generic.IEnumerable<System.String>.GetEnumerator' on type '' failed.
```